core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenTaskSource.java

```diff
@@ -464,15 +465,15 @@ private synchronized void processNext()
                 return;
             }
             verify(splitLoadingFuture == null || splitLoadingFuture.isDone(), "splitLoadingFuture is still running");
-            long start = System.currentTimeMillis();
+            long start = System.nanoTime();
```

Comment:
Possible issue with `master` branch. Current consumer expects nanos.

Reply:
Extract to separate commit please as a fix. You could also make a seprate PR